src/django_components/tag_formatter.py

```diff
@@ -0,0 +1,269 @@
+    ```django
+    {% #component "mycomp" abc=123 %}
+    ```
+    """
+
+    def format_block_start_tag(self, name: str) -> str:
+        return "component"
+
+    def format_block_end_tag(self, name: str) -> str:
+        return "endcomponent"
+
+    def format_inline_tag(self, name: str) -> str:
+        return "#component"
+
+    def parse_block_start_tag(self, tokens: List[str]) -> Tuple[str, List[str]]:
+        if tokens[0] != "component":
+            raise TemplateSyntaxError(
+                f"ComponentTagFormatter: Component block start tag parser received tag '{tokens[0]}',"
+                " expected 'component'"
+            )
+        return self._parse_start_or_inline_tag(tokens)
+
+    def parse_inline_tag(self, tokens: List[str]) -> Tuple[str, List[str]]:
+        if tokens[0] != "#component":
+            raise TemplateSyntaxError(
+                f"ComponentTagFormatter: Component block start tag parser received tag '{tokens[0]}',"
+                " expected '#component'"
+            )
+        return self._parse_start_or_inline_tag(tokens)
+
+    def _parse_start_or_inline_tag(self, tokens: List[str]) -> Tuple[str, List[str]]:
+        _, *comp_args = tokens
+
+        if not comp_args:
+            raise TemplateSyntaxError("ComponentTagFormatter: Component tag did not receive tag name")
+
+        # First arg is a kwarg, not a positional arg. Hence look for the "name" kwarg
+        # for component name.
+        if "=" in comp_args[0]:
+            comp_name = None
+            final_args = []
+            for kwarg in comp_args:
+                if not kwarg.startswith("name="):
+                    final_args.append(kwarg)
+                    continue
+
+                if comp_name:
+                    raise TemplateSyntaxError(
+                        f"ComponentTagFormatter: 'name' kwarg for component '{comp_name}'"
+                        " was defined more than once."
+                    )
+
+                # NOTE: We intentionally do NOT add to `final_args` here
+                # because we want to remove the the `name=` kwarg from args list
+                comp_name = kwarg[5:]
+        else:
+            comp_name = comp_args.pop(0)
+            final_args = comp_args
+
+        if not comp_name:
+            raise TemplateSyntaxError("Component name must be a non-empty quoted string, e.g. 'my_comp'")
+
+        if not is_str_wrapped_in_quotes(comp_name):
+            raise TemplateSyntaxError(f"Component name must be a string 'literal', got: {comp_name}")
+
+        # Remove the quotes
+        comp_name = resolve_string(comp_name)
+
+        return comp_name, final_args
+
+
+# TODO - Add tests
+# TODO - UPDATE DOCS FOR EVERYTHING - ShorthandTagFormatter, ComponentTagFormatter, TagFormatterABC, tag_formatter settings
+# TODO - DOCUMENT IN README HOW TAG_FORMATTER WORKS
+# TODO - DOCUMENT IN README HOW INLINE VS "BLOCK" components work
+class ShorthandTagFormatter(TagFormatterABC):
```

Comment:
This is the syntax similar to django-web-components or django-slippers, where the tag name is derived from the component name, e.g.:
```django
{% mycomp abc=123 %}
    {% fill "myfill" %}
        ...
    {% endfill %}
{% endmycomp %}
```

I included this as it seems as a recurring theme that some users want shorter syntax for components.